### PR TITLE
Only debounce the text field

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/v3-project-management/issues/250.

### CODE REVIEW

File: app/javascript/components/mentoring/Inbox.jsx [hunk 1]
```diff
@@ -9,13 +9,16 @@ import { usePaginatedRequestQuery } from '../../hooks/request-query'
 import { useIsMounted } from 'use-is-mounted'
 import { ResultsZone } from '../ResultsZone'
 import { useHistory } from '../../hooks/use-history'
-import { useDebounce } from '../../hooks/use-debounce'
 
 export function Inbox({ tracksRequest, sortOptions, ...props }) {
-  const { request, setCriteria, setOrder, setPage, setQuery } = useList(
-    props.discussionsRequest
-  )
-  const debouncedQuery = useDebounce(request.query, 500)
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/Inbox.jsx:18]: What's the difference between the old code and the new code?

Comment (iHiD) [reply app/javascript/components/mentoring/Inbox.jsx:18]: Previously everything was being routed via the debounce (including changing pagination-pages, changing tabs), etc. Now only the textual search goes via it.

File: app/javascript/components/mentoring/Inbox.jsx [hunk 2]
```diff
@@ -24,12 +27,22 @@ export function Inbox({ tracksRequest, sortOptions, ...props }) {
...
+    ['mentor-discussion-list', request.endpoint, request.query],
     request,
     isMountedRef
   )
 
-  useHistory({ pushOn: debouncedQuery })
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setRequestCriteria(criteria)
+    }, 500)
```
Comment (iHiD) [inline app/javascript/components/mentoring/Inbox.jsx]:
```suggestion
    }, 200)
```

File: app/javascript/components/mentoring/Inbox.jsx (hunk 1 above)
Comment (ErikSchierboom) [reply app/javascript/components/mentoring/Inbox.jsx:18]: Aha, okay!

**Pull request merged**